File: flutter-app/lib/maps/trip.dart
class Participant {
  final String id;
  final String name;
  final String role;
  bool reachedMeetingPoint;
  bool reachedDestination;

  Participant({
    required this.id,
    required this.name,
    required this.role,
    this.reachedMeetingPoint = false,
    this.reachedDestination = false,
  });
}

class Trip {
  final String name;
  final String destination;
  final List<Participant> participants;
  bool isSimulating;
  int simulationStep;

  Trip({
    required this.name,
    required this.destination,
    required this.participants,
    this.isSimulating = false,
    this.simulationStep = 0,
  });
}

File: website/types/trip.ts
export interface Location {
  lat: number;
  lng: number;
}

export interface Participant {
  id: string;
  name: string;
  location: Location;
  role?: ParticipantRole;
  status?: 'friend' | 'available' | 'selected';
  currentLocation?: Location; // For simulation
  reachedMeetingPoint?: boolean;
  reachedDestination?: boolean;
}

export enum ParticipantRole {
  LEADER = "Leader",
  MEMBER = "Member",
  TRAILER = "Trailer"
}

export interface Destination {
  name: string;
  location: Location;
}

export interface MeetingPoint {
  name?: string;
  location: Location;
  participants: Participant[];
}

export interface Route {
  name: string;
  from: Location;
  to: Location;
  coordinates: [number, number][];
  distance: number;
  duration: number;
  color: string;
  phase: 'meeting' | 'destination';
}

export interface Trip {
  id: string;
  name: string;
  destination: Destination;
  participants: Participant[];
  meetingPoints?: MeetingPoint[];
  routes?: Route[];
  phase?: 'meeting' | 'destination';
  createdAt: Date;
  isSimulating?: boolean;
  simulationStep?: number;
}